### ISSUE TITLE
Shared Routes inside tabs group not setting initial route

### ISSUE THREAD

**Issue by the-devbear** [expo/expo#27574]
### Minimal reproducible example

https://github.com/the-devbear/expo-router-mre

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

Hello 👋 
I'm not sure if this a bug or misconfiguration on my end.
But when I open the example app and navigate to the home/tabs screen and switch to the profile tab without opening the misc tab and click the button, I don't get the initial route for the misc tab.
When I then navigate back to the "Sign In" page and back to home and then open the misc tab once go to profile and click the link it works as expected.
Another thing is, that when I use deep linking with this:
`npx uri-scheme open exp://192.168.2.132:8081/--/\(misc\)/shared --ios`
it also works as expected and I get the back button at the header.
Would love some feedback.

TLDR to Reproduce:
Sign In ➡️ Home ➡️ Profile ➡️  Press Link ➡️ On Misc but no back button
Sign In ➡️ Home ➡️ Misc ➡️ Profile ➡️ Press Link ➡️ Works as aspected.

### Environment

```
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.1.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.6.7 - ~/.nvm/versions/node/v18.17.1/bin/npm
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Xcode: 15.0/15A240d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.11 => 50.0.11
      expo-router: ~3.4.8 => 3.4.8
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.73.4 => 0.73.4
      react-native-web: ~0.19.6 => 0.19.10
    npmGlobalPackages:
      eas-cli: 7.4.0
      expo-cli: 6.3.10
    Expo Workflow: managed
   ```

**Comment by marklawlor** [comment]
Fixed by #27485
